src/client_backend/triton_c_api/triton_loader.cc

```diff
@@ -1043,22 +1048,25 @@ TritonLoader::GetOutputs(
 
     std::string data_type{datatype};
     std::vector<uint8_t> data_copy;
-    // The first 4 bytes of BYTES data is a 32-bit integer to indicate the size
-    // of the rest of the data (which we already know based on byte_size). It
-    // should be ignored here, as it isn't part of the actual response
     if (data_type == "BYTES" && byte_size >= 4) {
-      data_copy.reserve(byte_size - 4);
-      std::copy(
-          static_cast<const uint8_t*>(base) + 4,
-          static_cast<const uint8_t*>(base) + byte_size,
-          std::back_inserter(data_copy));
-    } else {
-      data_copy.reserve(byte_size);
-      std::copy(
-          static_cast<const uint8_t*>(base),
-          static_cast<const uint8_t*>(base) + byte_size,
-          std::back_inserter(data_copy));
+      base = static_cast<const uint8_t*>(base) + 4;
+      byte_size -= 4;
+    }
+#ifdef TRITON_ENABLE_GPU
+    CUDARuntimeLibraryManager cuda_manager;
```

Comment:
Instantiating the library object like this will call dlopen each time. We don't want the library loaded each time GetOutputs() is called. Can you make this a data member like how it's used in InferDataManagerShm?:

https://github.com/triton-inference-server/perf_analyzer/blob/30ccf3201c450b7108faae71403e409e0b91481a/src/infer_data_manager_shm.h#L167

Reply:
So, dlopen won't open the same .so file twice if it hasn't been unloaded: it's reference counted. The waste isn't as bad as you think that is.